Issue 30 uri schema

evtimo opened the pull request.

evtimo committed:
storage-net-uri-schema

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
storage-net-uri-schema

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
new URI()

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
fixing

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java:
```diff
@@ -72,7 +81,7 @@ public T create(
 			final int batchSize
 	) throws OmgShootMyFootException, InterruptedException {
 		return (T) new com.emc.mongoose.storage.driver.pravega.PravegaStorageDriver<I, O>(
-				Constants.DEFAULT_CONTROLLER_URI, stepId, dataInput, storageConfig, verifyFlag, batchSize
```
the URI schema value should be under the "storageConfig" structure with the relative path "net-uri-schema", no additional argument is needed

akurilov commented on src/main/resources/config-schema-storage-net.json:
```diff
@@ -11,6 +11,7 @@
 			"tcpNoDelay": "boolean",
 			"timeoutMilliSec": "int32",
 			"ssl": "boolean",
+			"storage-net-uri-schema":"string",
```
the option is already inside the "storage" and "net" blocks, should be instead:
```
            "uri" : {
                "schema": "string"
            }
```

akurilov commented on src/main/resources/config/defaults-storage-net.json:
```diff
@@ -15,6 +15,7 @@
 			// Currently only HTTPS implementation is available.
 			// Have no effect if configured storage type is filesystem.
 			"ssl": false,
+			"storage-net-uri-schema": "tcp",
```
the option is already inside the "storage" and "net" blocks, should be instead:
```
            "uri": {
                "schema": "tcp"
            }
```

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -55,13 +55,13 @@ public PravegaStorageDriver(
 			throws OmgShootMyFootException {
 		super(testStepId, dataInput, storageConfig, verifyFlag, batchSize);
 		this.uriSchema = uriSchema;
```
should be:
```
this.uriSchema = storageConfig.stringVal("net-uri-schema");
```

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java:
```diff
@@ -51,6 +51,15 @@ public final String id() {
 		return NAME;
 	}
 
+	private final String uriSchema()  {
```
`SCHEMA_PROVIDER` is a configuration validation schema. This doesn't relate to the URI schema. I don't think that this method is needed

evtimo committed:
fixing after review

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
fixing after review

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

veronikaKochugova commented:
Please resolve the conflict

veronikaKochugova reviewed (changes requested):
Please resolve the conflict

evtimo committed:
resolve conflicts

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
resolve conflicts

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo commented:
tests fail because of missing parts of code from other new branches.
compilation without errors.

evtimo committed:
submitEventReadOperation

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
revert unused imports

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Evgeny <zheny.tim@gmail.com>

evtimo committed:
Merge branch 'master' of https://github.com/emc-mongoose/mongoose-storage-driver-pravega

Signed-off-by: Evgeny <zheny.tim@gmail.com>

# Conflicts:
#	src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

evtimo committed:
Merging

evtimo committed:
Merge remote-tracking branch 'origin/issue-30-uriSchema' into issue-30-uriSchema

# Conflicts:
#	src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java
#	src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java
#	src/test/java/com/emc/mongoose/storage/driver/pravega/integration/CommonTest.java
#	src/test/java/com/emc/mongoose/storage/driver/pravega/integration/DataOperationsTest.java

evtimo commented:
Difficult to resolve conflicts after adoption, issue solved in another branch
